### PR TITLE
Refs task #3145 create ui Home

### PR DESCRIPTION
**Pull request by tuanda5353**
### Summary:
- [Feature] Init Home screen
### Link Issue:
- ....https://edu-redmine.framgia.vn/issues/3145

### CODE REVIEW

**Comment by tuanda5353** [comment]
ready

File: app/src/main/java/framgia/com/ichat/screen/home/HomeActivity.java [hunk 1]
```diff
@@ -1,15 +1,61 @@
...
+import android.view.MenuInflater;
+import android.view.MenuItem;
 
 import framgia.com.ichat.R;
+import framgia.com.ichat.screen.base.BaseActivity;
 
-public class HomeActivity extends AppCompatActivity {
+public class HomeActivity extends BaseActivity implements android.support.v7.widget.SearchView.OnQueryTextListener {
+    private Toolbar mToolbar;
+    private android.support.v7.widget.SearchView mSearchView;
```
Comment (DoanVanToan) [inline app/src/main/java/framgia/com/ichat/screen/home/HomeActivity.java]: import android.support.v7.widget.SearchView
'
```suggestion
    private SearchView mSearchView;
```

File: app/src/main/java/framgia/com/ichat/screen/home/HomeActivity.java [hunk 2]
```diff
@@ -1,15 +1,61 @@
...
+    protected void initData(Bundle savedInstanceState) {
+
+    }
+
+    @Override
+    public boolean onCreateOptionsMenu(Menu menu) {
+        MenuInflater inflater = getMenuInflater();
+        inflater.inflate(R.menu.activity_home, menu);
+        MenuItem searchItem = menu.findItem(R.id.search_home);
+        mSearchView = (android.support.v7.widget.SearchView) searchItem.getActionView();
```
Comment (DoanVanToan) [inline app/src/main/java/framgia/com/ichat/screen/home/HomeActivity.java]: tương tự 
```suggestion
        mSearchView = (SearchView) searchItem.getActionView();
```

File: app/src/main/java/framgia/com/ichat/screen/home/HomeActivity.java [hunk 3]
```diff
@@ -1,15 +1,61 @@
...
+import android.view.MenuItem;
 
 import framgia.com.ichat.R;
+import framgia.com.ichat.screen.base.BaseActivity;
 
-public class HomeActivity extends AppCompatActivity {
+public class HomeActivity extends BaseActivity implements android.support.v7.widget.SearchView.OnQueryTextListener {
+    private Toolbar mToolbar;
+    private android.support.v7.widget.SearchView mSearchView;
+    private TabLayout mTabLayout;
```
Comment (DoanVanToan) [inline app/src/main/java/framgia/com/ichat/screen/home/HomeActivity.java]: keep tablayout, viewpager, and searchview is local variable

File: app/src/main/java/framgia/com/ichat/screen/home/PagerAdapterHome.java [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+import android.support.v4.app.FragmentManager;
+import android.support.v4.app.FragmentPagerAdapter;
+
+import framgia.com.ichat.screen.privateroom.FragmentPrivateRoom;
+import framgia.com.ichat.screen.useronline.FragmentUserOnline;
+
+public class PagerAdapterHome extends FragmentPagerAdapter {
+    private static final int FRAGMENT_PRIVATE_ROOM = 0;
+    private static final int FRAGMENT_USER_ONLINE = 1;
+    private String mTabTitles[];
```
Comment (DoanVanToan) [inline app/src/main/java/framgia/com/ichat/screen/home/PagerAdapterHome.java]: khai báo 
```suggestion
    private String[] mTitles;
```

File: app/src/main/java/framgia/com/ichat/screen/home/PagerAdapterHome.java [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+    public PagerAdapterHome(FragmentManager fm, String[] tabTitles) {
```
Comment (DoanVanToan) [inline app/src/main/java/framgia/com/ichat/screen/home/PagerAdapterHome.java]:
```suggestion
    public PagerAdapterHome(FragmentManager fm, String[] titles) {
```

File: app/src/main/res/layout/tb_home.xml [hunk 6]
```diff
@@ -0,0 +1,23 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (DoanVanToan) [inline app/src/main/res/layout/tb_home.xml]: rename to toolbar_home

File: app/src/main/java/framgia/com/ichat/screen/useronline/FragmentUserOnline.java [hunk 7]
```diff
@@ -0,0 +1,25 @@
+package framgia.com.ichat.screen.useronline;
+
+import android.os.Bundle;
+import android.support.v7.widget.RecyclerView;
+
+import framgia.com.ichat.R;
+import framgia.com.ichat.screen.base.BaseFragment;
+
+public class FragmentUserOnline extends BaseFragment {
+    private RecyclerView mRecyclerView;
```
Comment (DoanVanToan) [inline app/src/main/java/framgia/com/ichat/screen/useronline/FragmentUserOnline.java]: keep it is localvảriable

File: app/src/main/java/framgia/com/ichat/screen/privateroom/FragmentPrivateRoom.java [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+
+import android.os.Bundle;
+import android.support.design.widget.FloatingActionButton;
+import android.support.v7.widget.RecyclerView;
+
+import framgia.com.ichat.R;
+import framgia.com.ichat.screen.base.BaseFragment;
+
+public class FragmentPrivateRoom extends BaseFragment {
+    private RecyclerView mRecyclerView;
```
Comment (DoanVanToan) [inline app/src/main/java/framgia/com/ichat/screen/privateroom/FragmentPrivateRoom.java]: keep it is local variable

File: app/src/main/java/framgia/com/ichat/screen/home/PagerAdapterHome.java [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+        super(fm);
+        this.mTabTitles = tabTitles;
+    }
+
+    @Override
+    public Fragment getItem(int i) {
+        switch (i) {
+            case FRAGMENT_PRIVATE_ROOM:
+                return new FragmentPrivateRoom();
```
Comment (DoanVanToan) [inline app/src/main/java/framgia/com/ichat/screen/home/PagerAdapterHome.java]: e có thể tạo khởi tạo cho fragment bằng cáhc như sau
https://medium.com/@azjkjensen/using-the-newinstance-pattern-in-kotlin-e40c1b4ba1ef

File: app/src/main/java/framgia/com/ichat/screen/home/PagerAdapterHome.java [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+            case FRAGMENT_USER_ONLINE:
```
Comment (DoanVanToan) [inline app/src/main/java/framgia/com/ichat/screen/home/PagerAdapterHome.java]:
```suggestion
            case FRAGMENT_USER_ONLINE:
            default:
                return new FragmentUserOnline();
```

**Comment by DoanVanToan** [comment]
commented

**Comment by tuanda5353** [comment]
ready

File: app/src/main/java/framgia/com/ichat/screen/home/HomePagerAdapter.java [hunk 11]
```diff
@@ -0,0 +1,44 @@
...
+import android.support.v4.app.FragmentPagerAdapter;
+
+import framgia.com.ichat.screen.privateroom.PrivateRoomFragment;
+import framgia.com.ichat.screen.publicroom.PublicRoomFragment;
+import framgia.com.ichat.screen.useronline.UserOnlineFragment;
+
+public class HomePagerAdapter extends FragmentPagerAdapter {
+    private static final int PRIVATE_ROOM_FRAGMENT = 0;
+    private static final int USER_ONLINE_ROOM = 1;
+    private String mTitles[];
```
Comment (DoanVanToan) [inline app/src/main/java/framgia/com/ichat/screen/home/HomePagerAdapter.java]:
```suggestion
    private String[] mTitles;
```

File: app/src/main/java/framgia/com/ichat/screen/privateroom/PrivateRoomFragment.java [hunk 12]
```diff
@@ -0,0 +1,36 @@
...
+import android.support.v7.widget.RecyclerView;
+import android.view.View;
+
+import framgia.com.ichat.R;
+import framgia.com.ichat.screen.base.BaseFragment;
+
+public class PrivateRoomFragment extends BaseFragment implements View.OnClickListener {
+    public static PrivateRoomFragment newInstance() {
+        PrivateRoomFragment fragment = new PrivateRoomFragment();
+        return fragment;
```
Comment (DoanVanToan) [inline app/src/main/java/framgia/com/ichat/screen/privateroom/PrivateRoomFragment.java]:
```suggestion
        return new PrivateRoomFragment();;
```

File: app/src/main/java/framgia/com/ichat/screen/publicroom/PublicRoomFragment.java [hunk 13]
```diff
@@ -0,0 +1,28 @@
...
+
+import android.os.Bundle;
+
+import framgia.com.ichat.R;
+import framgia.com.ichat.screen.base.BaseFragment;
+
+public class PublicRoomFragment extends BaseFragment {
+    public static PublicRoomFragment newInstance() {
+        PublicRoomFragment fragment = new PublicRoomFragment();
+        return fragment;
```
Comment (DoanVanToan) [inline app/src/main/java/framgia/com/ichat/screen/publicroom/PublicRoomFragment.java]:
```suggestion
        return new PublicRoomFragment();;
```

File: app/src/main/java/framgia/com/ichat/screen/signup/SignUpActivity.java [hunk 14]
```diff
@@ -70,6 +71,16 @@ public void requestFocusPassword() {
         mEditPassword.requestFocus();
     }
 
+    @Override
+    public void showToastCreateAccountSuccess() {
```
Comment (DoanVanToan) [inline app/src/main/java/framgia/com/ichat/screen/signup/SignUpActivity.java]:
```suggestion
    public void onCreateAccountSuccess() {
```

File: app/src/main/java/framgia/com/ichat/screen/signup/SignUpActivity.java [hunk 15]
```diff
@@ -70,6 +71,16 @@ public void requestFocusPassword() {
...
+        Toast.makeText(this, R.string.msg_create_account_success, Toast.LENGTH_SHORT).show();
+    }
+
+    @Override
+    public void showToastCreateAccountFailed() {
```
Comment (DoanVanToan) [inline app/src/main/java/framgia/com/ichat/screen/signup/SignUpActivity.java]:
```suggestion
    public void onCreateAccountFailed() {
```

File: app/src/main/java/framgia/com/ichat/screen/useronline/UserOnlineFragment.java [hunk 16]
```diff
@@ -0,0 +1,29 @@
...
+import android.os.Bundle;
+import android.support.v7.widget.RecyclerView;
+
+import framgia.com.ichat.R;
+import framgia.com.ichat.screen.base.BaseFragment;
+
+public class UserOnlineFragment extends BaseFragment {
+    public static UserOnlineFragment newInstance() {
+        UserOnlineFragment fragment = new UserOnlineFragment();
+        return fragment;
```
Comment (DoanVanToan) [inline app/src/main/java/framgia/com/ichat/screen/useronline/UserOnlineFragment.java]:
```suggestion
        return new UserOnlineFragment();;
```

File: app/src/main/java/framgia/com/ichat/screen/home/HomeActivity.java [hunk 17]
```diff
@@ -1,15 +1,59 @@
...
+import android.view.MenuInflater;
+import android.view.MenuItem;
+import android.widget.SearchView;
 
 import framgia.com.ichat.R;
+import framgia.com.ichat.screen.base.BaseActivity;
 
-public class HomeActivity extends AppCompatActivity {
+public class HomeActivity extends BaseActivity implements SearchView.OnQueryTextListener {
+    private Toolbar mToolbar;
```
Comment (DoanVanToan) [inline app/src/main/java/framgia/com/ichat/screen/home/HomeActivity.java]: keep toolbar is local variable

**Comment by DoanVanToan** [comment]
commented

**Comment by DoanVanToan** [comment]
commented

**Comment by tuanda5353** [comment]
ready

**Review by DoanVanToan — APPROVED** [review]

**Pull request merged**